lib/sles4sap/sap_deployment_automation_framework/inventory_tools.pm

```diff
@@ -0,0 +1,220 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+package sles4sap::sap_deployment_automation_framework::inventory_tools;
+
+use warnings;
+use strict;
+use YAML::PP;
+use testapi;
+use Exporter qw(import);
+use Carp qw(croak);
+use sles4sap::sap_deployment_automation_framework::naming_conventions qw(deployer_private_key_path sut_private_key_path);
+use sles4sap::console_redirection;
+
+=head1 SYNOPSIS
+
+Library contains functions that handle SDAF inventory file.
+
+SDAF inventory yaml file example:
+QES_DB:
+  hosts:
+    dbhost01:
+      ansible_host        : 192.168.1.2
+      ansible_user        : someuser
+      ansible_connection  : ssh
+      connection_type     : key
+      virtual_host        : virtualhostname01
+      become_user         : root
+      os_type             : linux
+      vm_name             : somevmname01
+    dbhost02:
+      ansible_host        : 192.168.1.3
+      ansible_user        : someuser
+      ansible_connection  : ssh
+      connection_type     : key
+      virtual_host        : virtualhostname02
+      become_user         : root
+      os_type             : linux
+      vm_name             : somevmname02
+  vars:
+    node_tier             : hana
+    supported_tiers       : [hana, scs, pas]
+QES_SCS:
+  hosts:
+  vars:
+    node_tier             : scs
+    supported_tiers       : [scs, pas]
+QES_ERS:
+  hosts:
+  vars:
+    node_tier             : ers
+    supported_tiers       : [ers]
+
+=cut
+
+our @EXPORT = qw(
+  read_inventory_file
+  prepare_ssh_config
+  verify_ssh_proxy_connection
+);
+
+=head2 read_inventory_file
+
+    read_inventory_file($sap_inventory_file_path);
+
+Returns full path to an existing ansible inventory file
+
```

Comment:
Is this comment accurate? Seems to me it gets full path as INPUT and return data as OUTPUT.

Reply:
Oh yes... thanks. I fixed it to: 
```
Returns SDAF inventory file content in perl HASHREF format. 
```